smart_simulation/scale.py

```diff
@@ -61,95 +63,182 @@ def upsample(dataset, consumption_window_template: str) -> pd.DataFrame:
     return upsampled_data
 
 
-def create_weight_data(dataset: pd.DataFrame, arrival_template: str) -> pd.DataFrame:
+def calculate_arrival_windows(delivery_days_max: dict, freq: str) -> dict:
+    """
+    Calculate the arrival windows in number of time stamps for early, on-time, and late deliveries
+    Args:
+        delivery_days_max: Maximum days in early, on-time, late delivery window
+        freq: Frequency of the dataset utilized to calculate number of time stamps per day
+
+    Returns:
+        A dictionary of tuples as the range of time stamps for early, on-time, and late delivery windows
+    """
+    time_stamps_per_day = pd.Timedelta("1 day") / freq
```

Comment:
The type in the function for `freq` is listed as `str` I don't think that's right.

Reply:
You're correct, my mistake. I'm changing the input type to `pandas.Timedelta` to make it more understandable than the `frequency` format.